### PR TITLE
Optimize compose & DoMany

### PR DESCRIPTION
**Pull request by guusw**

### CODE REVIEW

**Comment by sinkingsugar** [comment]
Still very surprised that boost flat_map was so much slower

**Comment by sinkingsugar** [comment]
Btw the best optimization would be to recycle this https://github.com/fragcolor-xyz/shards/blob/1fa4e158c499f422ede2d46ec92720a16d936a4d/shards/core/runtime.cpp#L850 - should be easy to keep it in context

**Comment by guusw** [comment]
> Btw the best optimization would be to recycle this
> 
> https://github.com/fragcolor-xyz/shards/blob/1fa4e158c499f422ede2d46ec92720a16d936a4d/shards/core/runtime.cpp#L850
> - should be easy to keep it in context

Didn't even see that one, can fix that yes

**Comment by guusw** [comment]
Oh that's devel, nvm

File: shards/core/serialization.hpp [hunk 1]
```diff
@@ -151,6 +156,11 @@ struct Serialization {
 
     output.valueType = nextType;
 
+    // Uncomment for more detailed breakdown
```
Comment (sinkingsugar) [inline shards/core/serialization.hpp]: Maybe `#if 0` it instead?

File: shards/core/wire_doppelganger_pool.hpp [hunk 2]
```diff
@@ -41,63 +59,131 @@ template <typename T> struct WireDoppelgangerPool {
...
+    requires WireData<T>
+  {
+    ZoneScoped;
 
-      return value;
-    }
+    auto batch = acquireBatch(1);
+    return acquireFromBatch(batch, 0, composer, anything);
+
+    // pmr::SharedTempAllocator tempAlloc;
```
Comment (sinkingsugar) [inline shards/core/wire_doppelganger_pool.hpp]: Delete?

**Review by sinkingsugar — CHANGES_REQUESTED** [review]
Minor stuff

**Review by sinkingsugar — APPROVED** [review]

**Pull request merged**